### DISCUSSION TITLE
Mui, memory - when it worth it?

### DISCUSSION THREAD

**Discussion by ViacheslavK9** [olikraus/u8g2#2338] [Q&A] [answered]
Hi again, Oliver!

Just a 2 general questions regarding memory consumption:
1. When do you think it its worth to use Mui functionality?
Now i am using it on a Lora32u4 with 32 Kb of memory - already studying materials about "memory respect" code, because its 88% used for now.

2. can you give just a couple of general tips - which way better to design Muis in regard with (again) memory consumption.

Thanks!

**Comment by olikraus** [comment, marked as answer]
1.
If you just need a very very small menu system (like this: https://github.com/olikraus/u8g2/blob/master/sys/arduino/u8g2_page_buffer/IconMenu/IconMenu.ino), then probably MUI is not required. MUI has been designed to maximize reuse of code. So if you have multiple menues, the code doesn't increase much. For a new form, just a view bytes are required.

In general, the question is, what exactly occupies your memory. I personally suggest to look into the linker map file to check the memory consumption of other libs.
U8g2 (and MUI) are designed to minimize memory consumption, but this might not be true for other libs. 

MUI size estimation (SSD1306 HW I2C, AVR): 
MUICountDown requires 14222 Bytes
HelloWorld  requires 9884 Bytes

So from these examples MUI requires just 4.5KB. What will your other libs consume?

2.

 - Limit the number of fonts. Try to use transparent fonts with restricted number of chars (tr). If you need extra glyphs, consider to create a custom font.
 - Reuse MUI widgets as much as possible.

**Reply by ViacheslavK9** [reply]
Thanks, thats very helpful!
